client/src/pages/EditPage/view.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PageLayout from '@components/PageLayout';
+import { Redirect } from 'react-router-dom';
+import PublishTool from '@containers/PublishTool';
+
+class EditPage extends React.Component {
+  componentDidMount () {
+    const {asset, match, onHandleShowPageUri, setUpdateTrue, updateMetadata} = this.props;
+    onHandleShowPageUri(match.params);
+    setUpdateTrue();
+    if (asset) {
+      ['title', 'description', 'license', 'nsfw'].forEach(meta => updateMetadata(meta, asset.claimData[meta]));
+    }
+  }
+  componentWillUnmount () {
+    this.props.clearFile();
+  }
+  render () {
+    const { myChannel, asset } = this.props;
+    // redirect if user does not own this claim
+    if (
+      !myChannel || (
+        asset &&
+        asset.claimsData &&
+        asset.claimsData.channelName &&
+        asset.claimsData.channelName !== myChannel
+      )
+    ) {
+      return (<Redirect to={'/'} />);
```

Comment:
Do we need to redirect to login? Show an error? How would this happen to a user genuinely? 

(one guess: sign in in one tab, then sign in a second tab, and click edit from the first tab)

Reply:
I was only considering a nefarious user. I don't think this matters much; just needs to not render the edit page.